### PR TITLE
Feature: Scenes & Small Optimizations

### PR DESCRIPTION
**Pull request by dislogical**

### CODE REVIEW

File: source/graphics/shaders/shaders.d [hunk 1]
```diff
@@ -50,32 +37,27 @@ public enum ShaderUniform
...
+		ambientLight = new Shader( "AmbientLight", ambientlightVS, ambientlightFS, true );
+		directionalLight = new Shader( "DirectionalLight", directionallightVS, directionallightFS, true );
+		pointLight = new Shader( "PointLight", pointlightVS, pointlightFS, true );
+		userInterface = new Shader( "UserInterface", userinterfaceVS, userinterfaceFS, true );
+
 		foreach( file; FilePath.scanDirectory( FilePath.Resources.Shaders, "*.fs.glsl" ) )
 		{
 			// Strip .fs from file name
 			string name = file.baseFileName[ 0..$-3 ];
-			// if statement hitler
```
Comment (brooklynlittell) [inline source/graphics/shaders/shaders.d:65]: Thank you.

**Comment by ericchristenson** [comment]
I'm happy as of my additions to it.

**Comment by brooklynlittell** [comment]
:+1:

**Comment by wozniakty** [comment]
:+1: looks like Travis is mad we haven't updated sample game yet.

**Comment by dislogical** [comment]
Weird, but it looks like Travis is wrong. [The last build on this branch](https://travis-ci.org/Circular-Studios/Dash/builds/22146459) succeeded.

**Comment by brooklynlittell** [comment]
Nah, we just need to merge Sample-Game.

**Comment by dislogical** [comment]
Exactly. Because it thought it was building on Develop, it failed to checkout `feature/Scenes` in Sample like it should have.

Are we good to merge?

**Comment by brooklynlittell** [comment]
Yeah, we've all looked at it.

**Pull request merged**